static/src/javascripts/projects/common/modules/experiments/tests/recommended-for-you.js

```diff
@@ -112,45 +140,71 @@ define([
-                return imageFromTemplate(item.headerImage.urlTemplate);
-            } else if (item.headerVideo) {
-                return imageFromTemplate(item.headerVideo.stillImage.urlTemplate);
-            } else {
-                return null;
-            }
-        }
-
-        function itemFromRecommendationItem(item) {
-            return {
-                'id': item.item.id,
-                'imageUrl': imageUrlFromItem(item.item),
-                'title': item.item.title,
-                'standFirst': item.item.standFirst
-            };
-        }
-
         function setupComponentAttentionTracking(trackingCode) {
             require(['ophan/ng'], function (ophan) {
                 ophan.trackComponentAttention(trackingCode, $recommendedForYouSection[0]);
             });
         }
 
         function insertSection(items) {
+            var $oldSection = $recommendedForYouSection;
+
             $recommendedForYouSection = $.create(template(recommendedForYouTemplate, {
                 profileIcon: svg(profileIcon, ['rounded-icon', 'rfy-profile-icon', 'control__icon-wrapper']),
-                guardianLogo: svg(guardianLogo),
                 items: items
             }));
 
+            if ($oldSection != null) {
+                return fastdom.write(function() {
+                    $oldSection.replaceWith($recommendedForYouSection);
+                    setupComponentAttentionTracking('recommended-for-you_user-history');
+                });
+            } else {
+                return fastdom.write(function() {
+                    $recommendedForYouSection.insertBefore($opinionSection);
+                    setupComponentAttentionTracking('recommended-for-you_user-history');
+                });
+            }
+        }
+
+        function hasGivenFeedback() {
+            return storage.local.get(cachedRecommendationsEnabledKey) != null;
+        }
+
+        function shouldShowRecommendations() {
+            return !!storage.local.get(cachedRecommendationsEnabledKey);
+        }
+
+        function registerFeedback(showRecommendations) {
+            storage.local.set(cachedRecommendationsEnabledKey, showRecommendations);
+        }
+
+        function insertOnBoardingSection() {
+            $recommendedForYouSection = $.create(template(recommendedForYouOptInTemplate, {
+                profileIcon: svg(profileIcon, ['rounded-icon', 'rfy-profile-icon', 'control__icon-wrapper']),
+                rightArrowIcon: svg(rightArrowIcon, ['i-right']),
+                guardianLogo: svg(guardianLogo)
+            }));
+
             return fastdom.write(function() {
                 $recommendedForYouSection.insertBefore($opinionSection);
+                $('.js-feedback-button-yes', $recommendedForYouSection[0]).each(function(el) {
+                    bean.on(el, 'click', function () {
+                        registerFeedback(true);
+                        $('.js-feedback', $recommendedForYouSection[0]).html(
```

Comment:
populateRecommendationsContainer() is async so might be slow or fail, this message gives the user feedback on their action immediately.